src/D2L.Bmx/Authenticate.cs

```diff
@@ -55,18 +60,56 @@ internal class Authenticator {
 				PromptMfaInput( selectedMfa.Name );
 				throw new NotImplementedException();
 			}
-			authenticatedState = await oktaApi.AuthenticateChallengeMfaOktaAsync( authState, selectedMfaIndex - 1, mfaInput );
+			sessionToken = await oktaApi.AuthenticateChallengeMfaOktaAsync( authState, selectedMfaIndex - 1, mfaInput );
 		} else if( authState.OktaSessionToken is not null ) {
-			authenticatedState = new OktaAuthenticatedState( true, authState.OktaSessionToken );
+			sessionToken = authState.OktaSessionToken;
 		}
 
-		if( authenticatedState is not null ) {
-			return authenticatedState;
+		if( !string.IsNullOrEmpty( sessionToken ) ) {
+			var sessionResp = await oktaApi.CreateSessionOktaAsync( new SessionOptions( sessionToken ) );
+			oktaApi.AddSession( sessionResp.Id );
+			CacheOktaSession( user, org, sessionResp.Id, sessionResp.ExpiresAt );
+			return new( true, sessionResp.Id );
```

Comment:
could be nice to name the argument here and other calls to this constructor.
literal values are almost always good candidates for named arguments